presto-geospatial/src/main/java/com/facebook/presto/geospatial/BingTileFunctions.java

```diff
@@ -0,0 +1,314 @@
+        int tileX = (int) clip(x * mapSize + 0.5, 0, mapSize - 1);
+        int tileY = (int) clip(y * mapSize + 0.5, 0, mapSize - 1);
+        return new BingTile(tileX / TILE_PIXELS, tileY / TILE_PIXELS, zoomLevel);
+    }
+
+    private static OGCGeometry tileToPolygon(BingTile tile)
+    {
+        Point upperLeftCorner = tileXYToLatitudeLongitude(tile.getX(), tile.getY(), tile.getZoomLevel());
+        Point lowerRightCorner = tileXYToLatitudeLongitude(tile.getX() + 1, tile.getY() + 1, tile.getZoomLevel());
+
+        Polyline boundary = new Polyline();
+        boundary.startPath(upperLeftCorner);
+        boundary.lineTo(lowerRightCorner.getX(), upperLeftCorner.getY());
+        boundary.lineTo(lowerRightCorner);
+        boundary.lineTo(upperLeftCorner.getX(), lowerRightCorner.getY());
+
+        final Polygon polygon = new Polygon();
+        polygon.add(boundary, false);
+
+        return OGCGeometry.createFromEsriGeometry(polygon, null);
+    }
+
+    /**
+     * @return true if the geometry and its bounding box match
+     */
+    private static boolean isBoundingBox(OGCGeometry geometry, Envelope envelope)
+    {
+        if (geometry instanceof OGCPoint) {
+            return true;
+        }
+
+        if (!(geometry instanceof OGCPolygon)) {
+            return false;
+        }
+
+        OGCPolygon polygon = (OGCPolygon) geometry;
+        if (polygon.numInteriorRing() > 0) {
+            return false;
+        }
+
+        final MultiVertexGeometry multiVertex = (MultiVertexGeometry) polygon.getEsriGeometry();
+        if (multiVertex.getPointCount() != 4) {
+            return false;
+        }
+
+        Set<Point> corners = new HashSet<>();
+        corners.add(new Point(envelope.getXMin(), envelope.getYMin()));
+        corners.add(new Point(envelope.getXMin(), envelope.getYMax()));
+        corners.add(new Point(envelope.getXMax(), envelope.getYMin()));
+        corners.add(new Point(envelope.getXMax(), envelope.getYMax()));
+
+        for (int i = 0; i < 4; i++) {
+            Point point = multiVertex.getPoint(i);
+            if (!corners.contains(point)) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    private static void checkZoomLevel(long zoomLevel)
+    {
+        checkCondition(zoomLevel > 0, ZOOM_LEVEL_TOO_SMALL);
+        checkCondition(zoomLevel <= MAX_ZOOM_LEVEL, ZOOM_LEVEL_TOO_LARGE);
+    }
+
+    private static void checkXYCoordinate(long coordinate, long zoomLevel)
+    {
+        checkCondition(coordinate >= 0 && coordinate < (1 << zoomLevel), "XY coordinates for a Bing tile at zoom level %s must be within [0, %s) range", zoomLevel, 1 << zoomLevel);
+    }
+
+    private static void checkQuadKey(@SqlType(StandardTypes.VARCHAR) Slice quadkey)
+    {
+        checkCondition(quadkey.length() > 0, QUAD_KEY_EMPTY);
+        checkCondition(quadkey.length() <= MAX_ZOOM_LEVEL, QUAD_KEY_TOO_LONG);
+    }
+
+    private static boolean validLatitude(double latitude)
```

Comment:
how about naming this method and the one below as `checkLatitude`/`checkLongitude` to make the naming consistent with the other check methods? Then they will call `checkCondition` internally.